crates/ika-core/src/dwallet_mpc/dwallet_mpc_service.rs

```diff
@@ -139,6 +145,29 @@ impl DWalletMPCService {
         }
     }
 
+    async fn update_network_keys(&mut self) {
+        match self.network_keys_receiver.has_changed() {
+            Ok(has_changed) => {
+                if has_changed {
+                    let new_keys = self.network_keys_receiver.borrow_and_update();
+                    for (key_id, key_data) in (*new_keys).iter() {
```

Comment:
```suggestion
                    for (key_id, key_data) in new_keys.iter() {
```
why do you need to deref?